src/components/YearSelection.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from 'reactstrap';
+
+const DropdownMenuStyle = {
+  height: '200px',
+  overflow: 'auto'
+};
+
+class YearSelection extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      dropdownOpen: false,
+      dropDownValue: 'Select a Year'
+    };
+  }
+
+  toggle() {
+    this.setState(prevState => ({
+      dropdownOpen: !prevState.dropdownOpen
+    }));
+  }
+
+  changeValue(e) {
+    this.setState({ dropDownValue: e.currentTarget.textContent });
+  }
+
+  render() {
+    let allYears = [];
+    for (let i = 1990; i < 2019; i++) {
+      allYears.push(i + 1);
+    }
+    let yearArray = allYears.map(year => (
+      <DropdownItem
+        id={year}
+        onClick={e => this.setState({ dropDownValue: e.target.id })}
+      >
+        {year}
+      </DropdownItem>
+    ));
+
+    const style = { width: 200, margin: 50 };
```

Comment:
Can this `style` constant be outside as well? For the same reason on line 37

Reply:
Moved outside